parser/prism/Translator.cc

```diff
@@ -878,7 +878,7 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 auto name = parser.resolveConstant(prismName);
                 sorbetName = gs.enterNameUTF8(name);
             } else { // An anonymous rest parameter, like `def foo(*)`
-                sorbetName = gs.freshNameUnique(core::UniqueNameKind::Parser, core::Names::star(), ++uniqueCounter);
+                sorbetName = gs.enterNameUTF8("*");
```

Comment:
Can use this shared instance:

```suggestion
                sorbetName = core::Names::star()
```

Compare: https://github.com/sorbet/sorbet/pull/8203/files#diff-b6ead23655c4ab14d663a4d1ad737bbf8634556bb7e2890b71d1606260c4b101